Dockerfile

```diff
@@ -6,6 +6,12 @@ WORKDIR /usr/src/epub-press
 COPY package.json package-lock.json ./
 RUN npm install --production
 
+# Automatically downloads and installs Calibre. To use Kindlegen instead, set ARG converter=kindlegen.
+ARG converter=calibre
+ENV CONVERSION_BACKEND=$converter
+RUN if [ "$converter" = "calibre" ] ; then apt-get update && apt-get install -y libgl1-mesa-glx | wget -nv -O- https://download.calibre-ebook.com/linux-installer.sh | sh /dev/stdin install_dir=calibre-bin/ isolated=y ; fi
```

Comment:
I actually considered the logic here for half a second, but probably not long enough; the reason is that it is the Calibre devs who suggest an OR at each stage in the installer. Now that I look at it again though, AND makes way more sense, short-circuiting if something breaks, as you said.

I'll make the edit and commit it now. I won't be able to test it myself, but if something goes wrong you should hopefully be able to catch it.

Reply:
Lulz - I've been spending too much time in JS land 😅. 

`|` is actually the PIPE operator in this case and it's feeding the output of `wget` into the install command.

I think this would work:

```
RUN if [ "$converter" = "calibre" ] ; then apt-get update && apt-get install -y libgl1-mesa-glx && wget -nv -O- https://download.calibre-ebook.com/linux-installer.sh | sh /dev/stdin install_dir=calibre-bin/ isolated=y ; fi
```

Reply:
Whoops; didn't catch that either. Maybe I did have something right in mind the first time. Changed that and committed it.